Rocket.Chat/Helpers/RealmAssorter.swift

```diff
@@ -0,0 +1,135 @@
+//
+//  RealmAssorter.swift
+//  Rocket.Chat
+//
+//  Created by Matheus Cardoso on 6/22/18.
+//  Copyright © 2018 Rocket.Chat. All rights reserved.
+//
+
+import RealmSwift
+
+typealias IndexPathsChanges = (deletions: [IndexPath], insertions: [IndexPath], modifications: [IndexPath])
+typealias IndexPathsChangesEvent = ((IndexPathsChanges) -> Void)
+
+class RealmAssorter<Object: RealmSwift.Object> {
+    typealias ResultsTransform = ((Results<Object>) -> Results<Object>)
+
+    // MARK: RealmSection
+
+    struct RealmSection {
+        let name: String
+        var objects: [Object]
+        let transform: ResultsTransform?
+
+        static func make(name: String, results: Results<Object>, transform: ResultsTransform?) -> RealmSection {
+            let results = transform?(results) ?? results
+            return RealmSection(name: name, objects: Array(results), transform: transform)
+        }
+    }
+
+    // MARK: RealmAssorter
+
+    private let primaryKey: String
+    private let results: Results<Object>
+    private var sections: [RealmSection]
+    private var token: NotificationToken?
+
+    init(realm: Realm, results: Results<Object>? = nil) {
+        guard let primaryKey = Object.primaryKey() else {
+            fatalError("Object must have a primary key")
+        }
+
+        self.primaryKey = primaryKey
+        self.results = results ?? realm.objects(Object.self)
+        self.sections = []
+        self.token = nil
+
+        self.token = self.results.observe { changes in
+            switch changes {
+            case .update(let values, let deletions, let insertions, let modifications):
+                self.handleUpdate(values: values, deletions: deletions, insertions: insertions, modifications: modifications)
```

Comment:
The logout crash is happening on the first line of `handleUpdate` method, I guess that it's being mistakenly held on memory (at the time of the log out it should be gone, or at least the token invalidated), but it's just a guess. I tried to capture `weak self` but no success.

Reply:
Fixed :+1: see if you can reproduce again, please